src/core/manager/mod.rs

```diff
@@ -726,11 +726,21 @@ impl<X: XConn> WindowManager<X> {
     }
 
     fn focus_screen(&mut self, sel: &Selector<'_, Screen>) -> &Screen {
+        let prev_wix = self.screens.focused().wix;
+
         let actions = self.screens.focus_screen(sel);
         if let Err(e) = self.handle_event_actions(actions) {
             (self.error_handler)(e);
         }
 
+        // update current workspace if it is different
+        let wix = self.screens.focused().wix;
```

Comment:
same here

Reply:
The event action also stops at the first if-test in `focus_workspace` here.